### PR TITLE
POC for a test that runs the client, and interacts with it.

### PR DESCRIPTION
**Pull request by modmuss50**
I think this would be better placed in its own project used to test fabric + api (with mods) in production like environment. 

Running this in CI should be possible as well with a little effort.

### CODE REVIEW

**Comment by modmuss50** [comment]
Here is a video of it in action: https://www.youtube.com/watch?v=NMjEE94nATM

File: src/test/groovy/net/fabricmc/loom/test/integration/ClientRunTest.groovy [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+class ClientRunTest extends Specification implements GradleProjectTestTrait, RobotTrait {
+	@Unroll
+	def "Run client (minecraft #version)"() {
+		setup:
+			def gradle = gradleProject(project: "minimalBase", version: PRE_RELEASE_GRADLE)
+			gradle.buildGradle << """
+                dependencies {
+                    minecraft "com.mojang:minecraft:${version}"
+                    mappings loom.layered() {
+					}
```
Comment (Juuxel) [inline src/test/groovy/net/fabricmc/loom/test/integration/ClientRunTest.groovy:50]: Uhh, mixed tabs and spaces?

Comment (modmuss50) [reply src/test/groovy/net/fabricmc/loom/test/integration/ClientRunTest.groovy:50]: Prob, I forgot to fix the formatter. I dont plan on merging this anytime soon tbh, it needs quite a bit of work to make it worth it.

**Comment by modmuss50** [comment]
Fabric API now does this, I dont think this should be a loom thing.

**Pull request closed without merging**